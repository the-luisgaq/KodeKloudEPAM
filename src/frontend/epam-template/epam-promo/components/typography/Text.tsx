import { createSkinComponent } from '@epam/uui-core';
import * as uui from '@epam/uui';

interface TextMods extends Pick<uui.TextProps, 'fontStyle' | 'fontWeight' | 'size' | 'lineHeight' | 'fontSize'> {
    /**
     * Defines component color.
     * @default 'gray80'
     */
    color?: 'blue' | 'green' | 'amber' | 'red' | 'white' | 'gray5' | 'gray50' | 'gray60' | 'gray80' | 'gray90' | uui.TextProps['color'];
}

/** Represents the properties of a Text component. */
export interface TextProps extends uui.TextCoreProps, TextMods {}

export const Text = createSkinComponent<uui.TextProps, TextProps>(
    uui.Text,
    (props) => {
        return ({
            color: props.color ?? 'gray80',
        });
    },
);
