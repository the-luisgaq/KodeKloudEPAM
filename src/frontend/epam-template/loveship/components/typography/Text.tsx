import { createSkinComponent } from '@epam/uui-core';
import * as uui from '@epam/uui';

type TextColors = 'sky' | 'grass' | 'sun' | 'fire' | 'white' | 'night50' | 'night300' | 'night400' | 'night500' | 'night600' | 'night700' | 'night800' | 'night900' | uui.TextProps['color'];

interface TextMods extends Pick<uui.TextProps, 'fontStyle' | 'fontWeight' | 'size' | 'lineHeight' | 'fontSize'> {
    /**
     * Defines component color.
     * @default 'night700'
     */
    color?: TextColors;
}

/** Represents the properties of the Text component. */
export interface TextProps extends uui.TextCoreProps, TextMods {}

export const Text = createSkinComponent<uui.TextProps, TextProps>(
    uui.Text,
    (props) => {
        return ({
            color: props.color ?? 'night700',
        });
    },
);
