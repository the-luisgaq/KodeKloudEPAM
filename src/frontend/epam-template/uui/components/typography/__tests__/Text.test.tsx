import React from 'react';
import { Text } from '../Text';
import { renderSnapshotWithContextAsync } from '@epam/uui-test-utils';

describe('Text', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(<Text>Test</Text>);
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with props', async () => {
        const tree = await renderSnapshotWithContextAsync(
            <Text color="primary" size="48" fontSize="24" lineHeight="30" onClick={ jest.fn } rawProps={ { style: { marginRight: '12' } } }>Test</Text>,
        );
        expect(tree).toMatchSnapshot();
    });
});
