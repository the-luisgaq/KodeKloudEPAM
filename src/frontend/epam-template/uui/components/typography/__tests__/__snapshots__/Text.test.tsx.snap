// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="root uui-text uui-size-36 lineHeight fontSize uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography-inline container"
  >
    Test
  </div>
</DocumentFragment>
`;

exports[`Text should be rendered correctly with props 1`] = `
<DocumentFragment>
  <div
    class="root uui-text uui-size-48 lineHeight fontSize uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography-inline container"
    style="--uui-text-font-size: 24px; --uui-text-line-height: 30px;"
  >
    Test
  </div>
</DocumentFragment>
`;
